src/app/shared/services/user-session.service.ts

```diff
@@ -29,18 +30,18 @@ export class UserSessionService implements OnDestroy {
 
   constructor(
     private authService: AuthService,
-    private http: CurrentUserHttpService,
+    private currentUserService: CurrentUserHttpService,
   ) {
-    this.subscription = this.authService.status$.pipe(
-      switchMap(auth => {
-        if (!auth.authenticated) return of<UserProfile|undefined>(undefined);
-        return this.http.getProfileInfo().pipe(
-          catchError(_e => EMPTY)
-        );
-      }),
-      distinctUntilChanged((p1, p2) => p1 === p2 || (!!p1 && !!p2 && p1.groupId === p2.groupId))
+    this.subscription = merge(
+      this.authService.status$.pipe(filter(auth => auth.authenticated)),
+      this.refresh$,
+    ).pipe(
+      switchMap(() => this.currentUserService.getProfileInfo().pipe(
+        catchError(_e => EMPTY)
+      )),
+      distinctUntilChanged((p1, p2) => p1 === p2 || (p1.groupId === p2.groupId))
```

Comment:
Have you tested it ? Because thanks to the `distinctUntilChanged` there is no change in profile if it is the same user.

Reply:
I have tested it, indeed currently the profile won't change which is not what I want BUT
If I change the distinct until changed condition here, I'll have to inspect every subscription to `userSession.session$`, I intended to open another PR for that.

Side note: The check `p1 === p2` is useless since `{} !== {}` because javascript :shrug:

Reply:
If this misbehaving was intended, you should have mentioned in the PR description so that I know I have to ignore that issue.

Reply:
I probably know better than you each usage. Actually, most should be straightforward but 1 or 2 may be more critical.

Reply:
I'll do the changes in that PR

Reply:
Fixed